app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -25,11 +26,64 @@ def bulk_destroy
         process_response @hosts.destroy_all
       end
 
+      api :PUT, "/hosts/bulk/build", N_("Build")
+      param_group :bulk_host_ids
+      param :reboot, :bool, N_("Reboot after build")
+      param :rebuild_configuration, :bool, N_("Rebuild configuration only")
+      def build
+        if Foreman::Cast.to_bool(params[:rebuild_configuration])
+          rebuild_config
+        else
+          reboot = Foreman::Cast.to_bool(params[:reboot])
+          manager = BulkHostsManager.new(hosts: @hosts)
+          missed_hosts = manager.build(reboot: reboot)
+          if missed_hosts.empty?
+            process_response(true, { :message => _("Built %{host_count} %{hosts}") % { :host_count => @hosts.count, :hosts => 'host'.pluralize(@hosts.count) }})
```

Comment:
This breaks translations because `host` or `hosts` will not be correctly translated. There are also languages which have a more complex plural system and `n_()` is the correct function to use. See https://projects.theforeman.org/projects/foreman/wiki/Translating#Extracting-strings as well:
```suggestion
            process_response(true, { :message => n_("Built %s host", "Built %s hosts", @hosts.count) % @hosts.count })
```

I don't know if `@hosts.count` will be cached or result in 2 queries, but the previous code already had that too.

Reply:
updated